darts/models/forecasting/forecasting_model.py

```diff
@@ -2220,3 +2220,40 @@ def _supress_generate_predict_encoding(self) -> bool:
     @property
     def extreme_lags(self):
         return (-1, 1, None, 0, 0)
+
+    def __str__(self):
```

Comment:
👍 Very neat, thanks!

Could you rename the method into `__repr__` and move it to the `ForecastingModel` base class?

Reply:
+1 for `__repr__`, however, it could also be beneficial to have `__str__` setup correctly as well I guess (the two are called in different circumstances). WDYT?

Reply:
I never really used `__repr__` but if I understood it's point correctly (pretty much just a more informative __str__), then my intuition would be to keep `__str__` as is and for `__repr__` provide all the key_value pairs of parameters including the defaults. This way users can both get simple model labels for graph legends when comparing hyper-parameters and get the entire set of model's parameters without having to delve into documentation to check their default values. Would that make sense?

Reply:
From what I understand the intention behind the magic methods is:

"""
The `__str__()` method is intended to return information in a readable format for humans to easily understand it. It is perfect for logging or displaying object information. While the `__repr__()` method's goal is to deliver a string representation of the object that may be used to replicate it, it is also designed to do the opposite. 
([source](https://www.educative.io/answers/what-is-the-difference-between-str-and-repr))
"""

So in our case, the `__str__` implementation in this PR can actually be used to create a new model object (it's doing what the `__repr__` is intended to be used for). 

From another [source](https://www.digitalocean.com/community/tutorials/python-str-repr-functions#what-s-the-difference-between-__str__-and-__repr__):
"""
If you don’t define a `__str__()` method for a class, **then the built-in object implementation calls the `__repr__()` method instead.**
"""

So I guess we would only have to define `__repr__` and not `__str__`. It is perfectly fine though like you did it, I was just curious myself what the best practices were for these two magic methods :)